### PR TITLE
Add ollama embeddings function

### PR DESCRIPTION
**Pull request by asmith26**
Following the docs [here](https://lancedb.github.io/lancedb/python/python/#lancedb.embeddings.openai.OpenAIEmbeddings) I've been trying to use ollama embedding via the OpenAI API interface, but unfortunately I couldn't get it to work (possibly related to https://github.com/ollama/ollama/issues/2416)

Given the popularity of ollama I thought it could be helpful to have a dedicated Ollama Embedding function in lancedb.

Very much welcome any thought on this or my code etc. Thanks!

### CODE REVIEW

**Comment by changhiskhan** [comment]
Does it work well for you locally? IIRC the ollama OpenAI api is only for LLM and doesn't work for the embedding endpoint yet. Has there been a recent new release?

**Comment by asmith26** [comment]
Hi @changhiskhan, thanks for the comments.

> Does it work well for you locally? 

The example I created for the docs https://github.com/lancedb/lancedb/pull/1263/files#diff-e4f8e4ca897425fa2aa42069418ab1b10a35b3b8233e30ab278d776ffb96fddcR222-R245 works well for me - I have briefly checked the first few vectors match the same output as running:

```bash
curl http://localhost:11434/api/embeddings -d '{
  "model": "nomic-embed-text",
  "prompt": "hello world"
}'
```

If you wanted to run the example (in this PR/link above), you should be able to install my lancedb fork with:

```bash
$ pip install "git+https://github.com/asmith26/lancedb.git@main#egg=lancedb&subdirectory=python"  

# If you need to setup ollama:
$ pip install ollama                              # to install ollama python client
$ curl -fsSL https://ollama.com/install.sh | sh   # to install ollama server
$ ollama serve &                                  # to start ollama server
$ ollama pull nomic-embed-text                    # to pull this embedding model
```


> IIRC the ollama OpenAI api is only for LLM and doesn't work for the embedding endpoint yet. Has there been a recent new release?

You're right, using the ollama OpenAI API for embeddings still doesn't seem to work (I think it's possibly waiting on https://github.com/ollama/ollama/issues/2416). Fortunately the ollama python client [does support embeddings](https://github.com/ollama/ollama-python?tab=readme-ov-file#embeddings), and that's what I'm using in this PR.

**Comment by AyushExel** [comment]
Thanks for the PR. Can you add a test like the ones in `test_embedding_slow.py` files and fix the CI failures?

**Comment by asmith26** [comment]
Thanks for the comment @AyushExel.

I've added `ollama` to `test_basic_text_embeddings` https://github.com/lancedb/lancedb/pull/1263/files#diff-0acbac5decf016cec4771fd47f40db345ce40ac24e2911fee5b77424e7274807R49 which tests the default settings, and this test passes for me locally (requires an Ollama server running locally). Would you like me to add anything else? 

---

For some reason the tests are failing on code unrelated to this PR (looks like `main` branch is failing with a similar problem, red x):
image
Looks like the failing test is related to full-text-search - having a quick read of https://lancedb.github.io/lancedb/fts/#phrase-queries-vs-terms-queries, possilby the underlying full-text-search engine used by lancedb (looks like Tantivy) has changed it's behaviour which is now causing this test to fail?

Thanks for your help! :)

File: python/python/lancedb/embeddings/ollama.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ) -> List["np.array"]:
+        """
+        Get the embeddings for the given texts
+
+        Parameters
+        ----------
+        texts: list[str] or np.ndarray (of str)
+            The texts to embed
+        """
+        # TODO retry, rate limit, token limit
```
Comment (AyushExel) [inline python/python/lancedb/embeddings/ollama.py:61]: Can remove this comment. Auto retry logic is now implemented. Handling token limit automatically is something we haven't thought of doing yet.

**Review by AyushExel — APPROVED** [review]

**Pull request merged**